R/ard_categorical.R

```diff
@@ -114,6 +114,9 @@ ard_categorical.data.frame <- function(data,
   # check inputs ---------------------------------------------------------------
   check_not_missing(variables)
   .check_no_ard_columns(data)
+  
+  check_no_na_factor_levels(data)
+  check_factor_has_levels(data)
```

Comment:
Here we are checking every column from `data`, but I think we can move these checks a bit lower in the script (after the selecting of by, strata, variables, etc), then we could run the checks only on the columns that are being used, e.g. `check_no_na_factor_levels(data[c(variables, by, strata)])`

Reply:
This comment applies to all the functions I think. One difference is that `ard_continuous()` and `ard_complex()` will not need to have the `variables` selected because we are not tabulating those.

Reply:
Factor checks have been moved lower in all appropriate `ard_*` functions, and only test the columns when the data is subset by `variables`, `by`, and `strata` (or just `by` and `strata` for `ard_continuous()` (no longer have the checks in place for `ard_complex()` as it calls `ard_continuous()`))

Reply:
Great, thank you!